src/gluonnlp/model/transformer.py

```diff
@@ -738,9 +990,9 @@ def forward(self, step_input, states, mask=None):  #pylint: disable=arguments-di
                                         axis=0, size=step_input.shape[0])
         steps = mx.nd.arange(step_input.shape[1], ctx=step_input.context)
         states.append(steps)
+        scaled_step_input = step_input * math.sqrt(step_input.shape[-1])
```

Comment:
To be consistent with Encoder. if self._scale_embed: is missing.

Reply:
Added.